### PR TITLE
Add K8sPod as a task target option

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
Add K8sPod as a task target option

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1050

## Follow-up issue
_NA_

### CODE REVIEW

File: protos/flyteidl/core/tasks.proto [hunk 1]
```diff
@@ -238,3 +239,22 @@ message DataLoadingConfig {
...
+// Defines a pod spec and additional pod metadata that is created when a task is executed.
+message K8sPod {
+   // Contains additional metadata for building a kubernetes pod.
+   K8sObjectMetadata metadata = 1;
+
+   // Defines the primary pod spec created when a task is executed.
+   google.protobuf.Struct pod_spec = 2;
+}
+
+// Metadata for building a kubernetes pod when a task is executed.
```
Comment (EngHabu) [inline protos/flyteidl/core/tasks.proto:252]: nit:
```suggestion
// Metadata for building a kubernetes object when a task is executed.
```

File: protos/flyteidl/core/tasks.proto [hunk 2]
```diff
@@ -238,3 +239,22 @@ message DataLoadingConfig {
     LiteralMapFormat format = 4;
     IOStrategy io_strategy = 5;
 }
+
+// Defines a pod spec and additional pod metadata that is created when a task is executed.
+message K8sPod {
+   // Contains additional metadata for building a kubernetes pod.
+   K8sObjectMetadata metadata = 1;
+
+   // Defines the primary pod spec created when a task is executed.
```
Comment (EngHabu) [inline protos/flyteidl/core/tasks.proto:248]: nit: want to add a link to the expected format here? maybe the python and go types?

**Review by EngHabu — APPROVED** [review]

File: protos/flyteidl/core/tasks.proto (hunk 1 above)
Comment (katrogan) [reply protos/flyteidl/core/tasks.proto]: done

File: protos/flyteidl/core/tasks.proto (hunk 2 above)
Comment (katrogan) [reply protos/flyteidl/core/tasks.proto:248]: sure, done

**Review by EngHabu — APPROVED** [review]

File: protos/flyteidl/core/tasks.proto (hunk 2 above)
Comment (kumare3) [reply protos/flyteidl/core/tasks.proto:248]: So we need to add K8sObject prefix? Can we just make it metadata. Not sure about the namespacing

Comment (katrogan) [reply protos/flyteidl/core/tasks.proto:248]: i think it's package namespaced, right? even in the context of this file i feel like metadata could be ambiguous

**Review by kumare3 — APPROVED** [review]

**Pull request merged**